src/main/kotlin/org/rust/lang/core/psi/RsPsiFactory.kt

```diff
@@ -261,7 +261,8 @@ class RsPsiFactory(
             ""
         } else {
             val parameterNames = typeParameterList.lifetimeParameterList.map { it.quoteIdentifier.text } +
-                typeParameterList.typeParameterList.map { it.name }
+                typeParameterList.typeParameterList.map { it.name } +
+                typeParameterList.constParameterList.map { it.name }
```

Comment:
Actually, it doesn't work as expected with `const_generics` feature where const and type parameters can be mixed

Reply:
I'll take a look after #7479

Reply:
Not sure #7479 is actually needed here. We recently with @mchernyavsky  introduced `RsGenericParameter` and `org.rust.lang.core.psi.ext.RsTypeParameterListKt#getGenericParameterList` to get single list of all generic arguments for such cases

Reply:
> Actually, it doesn't work as expected with `const_generics` feature where const and type parameters can be mixed

Do you have a reference for that feature? I have always believed that the restrictions on the order of parameters are not intended to be ever lifted, for readability reasons, same as we are always required to place lifetimes before type arguments.

Reply:
> Do you have a reference for that feature?

https://github.com/rust-lang/rust/pull/74953

Just try to compile the following code with nightly compiler
```rust
#![feature(const_generics)]

fn foo<const C: i32, T>() {}

fn main() {}
```

> I have always believed that the restrictions on the order of parameters are not intended to be ever lifted, for readability reasons, same as we are always required to place lifetimes before type arguments.

It's only about the order of const and type parameters. Lifetime parameters still should be placed first